assemblyline/templates/ingress.yaml

```diff
@@ -1,37 +1,37 @@
+# Generate self-signed certificate
 {{- if not .Values.tlsSecretName }}
-#  {{ $ca := genCA "svc-cat-ca" 3650 }}
-#  {{ $cn := .Values.domain }}
-  {{ $server := genSelfSignedCert .Values.configuration.ui.fqdn nil (list .Values.configuration.ui.fqdn) 365 }}
+{{ $server := genSelfSignedCert .Values.configuration.ui.fqdn nil (list .Values.configuration.ui.fqdn) 365 }}
 
 apiVersion: v1
 kind: Secret
 type: kubernetes.io/tls
 metadata:
   name: {{ .Release.Name }}-generated-cert
   labels:
-    app: assembyline
+    app: assemblyline
 data:
   tls.crt: {{ b64enc $server.Cert }}
   tls.key: {{ b64enc $server.Key }}
-{{ end }}
+{{- end }}
+
 ---
+
+# FRONTEND INGRESS (External)
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
-  name: {{ .Release.Name }}-ingress
+  name: {{ .Release.Name }}-frontend-ingress
   annotations:
     {{- range $key, $value := .Values.ingressAnnotations }}
     {{ $key }}: {{ $value | quote }}
     {{- end }}
+    appgw.ingress.kubernetes.io/backend-hostname: "frontend"
 spec:
-{{ if .Values.ingressClassName }}
-  ingressClassName: {{.Values.ingressClassName }}
-{{end}}
   tls:
     - hosts: [{{ .Values.configuration.ui.fqdn }}]
       secretName: {{ if .Values.tlsSecretName }} {{.Values.tlsSecretName }} {{else}} {{ .Release.Name }}-generated-cert {{end}}
   rules:
-    - host: {{ if .Values.ingressHost }} {{.Values.ingressHost }} {{else}} {{ .Values.configuration.ui.fqdn }} {{end}}
+    - host: {{ .Values.configuration.ui.fqdn | quote }}
```

Comment:
We'd probably want to retain the original use of an optional `ingressHost` with a default to the configuration's FQDN, just in case if someone is using this configuration.